plugins/source/gcp/codegen/recipes/aiplatform.go

```diff
@@ -0,0 +1,76 @@
+			Description:           "https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.customJobs#CustomJob",
+			RegisterServer:        pb.RegisterJobServiceServer,
+			UnimplementedServer:   &pb.UnimplementedJobServiceServer{},
+			MockImports:           []string{"google.golang.org/genproto/googleapis/cloud/location"},
+		},
+		{
+			SubService:            "indexes",
+			Struct:                &pb.Index{},
+			PrimaryKeys:           []string{ProjectIdColumn.Name, "name"},
+			NewFunction:           aiplatform.NewIndexClient,
+			ListFunction:          (&aiplatform.IndexClient{}).ListIndexes,
+			LocationsListFunction: (&aiplatform.JobClient{}).ListLocations,
+			RequestStruct:         &pb.ListIndexesRequest{},
+			ResponseStruct:        &pb.ListIndexesResponse{},
+			Description:           "https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.indexes#Index",
+			RegisterServer:        pb.RegisterIndexServiceServer,
+			UnimplementedServer:   &pb.UnimplementedIndexServiceServer{},
+			MockImports:           []string{"google.golang.org/genproto/googleapis/cloud/location"},
+			IgnoreInTestsColumns:  []string{"metadata"},
+		},
+	}
+
+	locationResources := []*Resource{}
+	for _, resource := range resources {
+		resource.ChildTable = true
+		resource.ProtobufImport = "cloud.google.com/go/aiplatform/apiv1/aiplatformpb"
+		resource.RequestStructFields = `Parent: parent.Item.(*location.Location).Name,`
+
+		locationResource := Resource{}
+		locationResource.SubService = resource.SubService + "_locations"
+		locationResource.Struct = &locationpb.Location{}
+		locationResource.PrimaryKeys = []string{ProjectIdColumn.Name, "name"}
+		locationResource.NewFunction = resource.NewFunction
+		locationResource.ListFunction = resource.LocationsListFunction
+		locationResource.RequestStruct = &locationpb.ListLocationsRequest{}
+		locationResource.ResponseStruct = &locationpb.ListLocationsResponse{}
+		locationResource.RequestStructFields = `Name: "projects/" + c.ProjectId,`
+		locationResource.Description = resource.Description
+		locationResource.RegisterServer = locationpb.RegisterLocationsServer
+		locationResource.UnimplementedServer = &locationpb.UnimplementedLocationsServer{}
+		locationResource.ProtobufImport = "google.golang.org/genproto/googleapis/cloud/location"
+		locationResource.Relations = []string{Caser.ToPascal(resource.SubService) + "()"}
+
+		locationResources = append(locationResources, &locationResource)
+	}
+
+	allResources := append(resources, locationResources...)
+	for _, resource := range allResources {
+		resource.Service = "aiplatform"
+		resource.Template = "newapi_list"
+		resource.MockTemplate = "newapi_list_grpc_mock"
+		resource.MockImports = append(resource.MockImports, "cloud.google.com/go/aiplatform/apiv1", "google.golang.org/api/option")
+		resource.ClientOptions = []string{`option.WithEndpoint("us-central1-aiplatform.googleapis.com:443")`}
```

Comment:
Setting the endpoint is required for this API https://github.com/googleapis/google-cloud-go/issues/6649#issuecomment-1242515615